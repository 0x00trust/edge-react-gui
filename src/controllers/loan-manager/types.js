// @flow

import { type BorrowEngine, type BorrowPlugin } from '../../plugins/borrow-plugins/types'
import { type LoanProgramEdge } from './store'

export type LoanAccount = {
  id: string,
  borrowPlugin: BorrowPlugin,
  borrowEngine: BorrowEngine,

  // Action Program Relationships
  programEdges: LoanProgramEdge[]
}

export type LoanAccountMap = {
  [walletId: string]: LoanAccount
}

export type LoanManager = {
  loanAccounts: LoanAccount[]
}
