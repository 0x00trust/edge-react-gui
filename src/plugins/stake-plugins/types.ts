// -----------------------------------------------------------------------------
// Stake Policy
// -----------------------------------------------------------------------------

import { EdgeCurrencyWallet } from 'edge-core-js'

export interface AssetId {
  pluginId: string
  currencyCode: string
}

// Defines what to display so that the user can identify the service provider
// of the staking policy.
export interface StakeProviderInfo {
  // Card subtitle
  displayName: string

  // CDN icon
  pluginId: string
  stakeProviderId: string
}

export interface StakePolicy {
  // Internal policy id, unique across all stake policies offered by Edge
  stakePolicyId: string

  // Some unique grouping information to display in the policy card to
  // differentiate between stake policies that share the same input stakeAssets
  stakeProviderInfo?: StakeProviderInfo

  // A percentage number representing the yield per year
  apy: number

  // The assets which can be earned
  rewardAssets: AssetId[]

  // The assets which must be staked
  stakeAssets: AssetId[]
}

// -----------------------------------------------------------------------------
// Change Quote
// -----------------------------------------------------------------------------
export interface ChangeQuoteRequest {
  action: 'stake' | 'unstake' | 'claim'
  stakePolicyId: string
  currencyCode: string
  nativeAmount: string
  wallet: EdgeCurrencyWallet
}

export interface QuoteAllocation {
  allocationType: 'stake' | 'unstake' | 'claim' | 'fee'
  pluginId: string
  currencyCode: string
  nativeAmount: string
}

export interface ChangeQuote {
  allocations: QuoteAllocation[]
  approve: () => Promise<void>
}

// -----------------------------------------------------------------------------
// Stake Position
// -----------------------------------------------------------------------------

export interface StakePositionRequest {
  stakePolicyId: string
  wallet: EdgeCurrencyWallet
}

export interface PositionAllocation {
  // The of asset for this allocation
  pluginId: string
  currencyCode: string
  // The of the allocation
  allocationType: 'staked' | 'unstaked' | 'earned'
  // Amount of the asset allocated
  nativeAmount: string
  /*
  A date/time when the allocation is available.
  Example: earned allocations with a future date are not available,
  but earned allocations with a past date are available to be earned.
  For some tokens (e.g. FIO), there is no earned allocation; rather there is
  an unstaked allocation which is locked until the date.
  */
  locktime?: Date
}

export interface StakePosition {
  allocations: PositionAllocation[]
  canStake: boolean
  canUnstake: boolean
  canClaim: boolean
}

// -----------------------------------------------------------------------------
// Stake Plugin
// -----------------------------------------------------------------------------

export interface StakePlugin {
  policies: StakePolicy[]
  fetchChangeQuote: (request: ChangeQuoteRequest) => Promise<ChangeQuote>
  fetchStakePosition: (request: StakePositionRequest) => Promise<StakePosition>
}
