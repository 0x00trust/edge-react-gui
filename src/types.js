/**
 * Created by paul on 8/16/17.
 */

// @flow

export interface ABCDenomination {
  name:string,
  multiplier:string,
  symbol:string|null
}

export interface ABCMetaToken {
  currencyCode:string,
  currencyName:string,
  denominations:Array<ABCDenomination>,
  contractAddress:string|null,
  symbolImage:string|null
}

export class GUIWallet {
  id:string
  type:string
  name:string
  primaryNativeBalance:string
  nativeBalances: { [currencyCode: string]: string }
  currencyNames: { [currencyCode: string]: string }
  currencyCode:string
  isoFiatCurrencyCode:string
  fiatCurrencyCode:string
  denominations:Array<ABCDenomination>
  allDenominations: { [currencyCode: string]: { [denomination: string]: ABCDenomination } }
  symbolImage:string
  metaTokens:Array<ABCMetaToken>
  sortIndex:number
  archived:boolean
  deleted:boolean
  constructor (
    id:string,
    type:string,
    name:string,
    primaryNativeBalance:string,
    nativeBalances: { [currencyCode: string]: string },
    currencyNames: { [currencyCode: string]: string },
    currencyCode:string,
    isoFiatCurrencyCode:string,
    fiatCurrencyCode:string,
    denominations:Array<ABCDenomination>,
    allDenominations: { [currencyCode: string]: { [denomination: string]: ABCDenomination } },
    symbolImage:string,
    metaTokens:Array<ABCMetaToken>,
    sortIndex:number,
    archived:boolean,
    deleted:boolean
  ) {
    this.id = id
    this.type = type
    this.name = name
    this.primaryNativeBalance = primaryNativeBalance
    this.nativeBalances = nativeBalances
    this.currencyNames = currencyNames
    this.currencyCode = currencyCode
    this.isoFiatCurrencyCode = isoFiatCurrencyCode
    this.fiatCurrencyCode = fiatCurrencyCode
    this.denominations = denominations
    this.allDenominations = allDenominations
    this.symbolImage = symbolImage
    this.metaTokens = metaTokens
    this.sortIndex = sortIndex
    this.archived = archived
    this.deleted = deleted
  }
}
