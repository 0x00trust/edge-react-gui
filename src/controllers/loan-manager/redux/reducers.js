// @flow

import { type Reducer, combineReducers } from 'redux'

import { type Action } from '../../../types/reduxTypes'
import { type LoanAccountMap } from '../types'

export type LoanManagerState = {
  +loanAccounts: LoanAccountMap
}

export const loanManager: Reducer<LoanManagerState, Action> = combineReducers({
  loanAccounts(state: LoanAccountMap = {}, action: Action): LoanAccountMap {
    switch (action.type) {
      case 'LOAN_MANAGER/SET_LOAN_ACCOUNT': {
        const { loanAccount } = action
        return {
          ...state,
          [loanAccount.id]: loanAccount
        }
      }
      case 'LOAN_MANAGER/DELETE_LOAN_ACCOUNT': {
        const { id } = action
        const { [id]: _, ...remainingState } = state
        return remainingState
      }
      case 'LOGOUT': {
        return {}
      }
      default:
        return state
    }
  }
})
