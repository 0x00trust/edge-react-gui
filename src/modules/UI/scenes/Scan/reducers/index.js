// @flow

import { addressModalVisible } from './addressModalVisible.js'
import { parsedUri } from './parsedUri.js'
import { scanEnabled } from './scanEnabled.js'
import { torchEnabled } from './torchEnabled.js'

export { addressModalVisible, scanEnabled, torchEnabled, parsedUri }
