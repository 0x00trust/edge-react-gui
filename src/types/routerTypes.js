// @flow

import { type EdgeCurrencyInfo, type EdgeCurrencyWallet, type EdgeTransaction, type JsonObject, type OtpError } from 'edge-core-js'
import * as React from 'react'
import * as Flux from 'react-native-router-flux'

import type { ExchangedFlipInputAmounts } from '../components/themed/ExchangedFlipInput.js'
import { type BorrowEngine, type BorrowPlugin } from '../plugins/borrow-plugins/types'
import { type FiatPluginEnterAmountResponse, type FiatPluginGetMethodsResponse } from '../plugins/gui/fiatPluginTypes.js'
import { type ChangeQuoteRequest, type StakePolicy, type StakePosition } from '../plugins/stake-plugins'
import { type GuiPlugin } from './GuiPluginTypes.js'
import {
  type CreateWalletType,
  type FeeOption,
  type FioConnectionWalletItem,
  type FioDomain,
  type FioRequest,
  type GuiFiatType,
  type GuiMakeSpendInfo,
  type GuiSwapInfo,
  type WcConnectionInfo
} from './types.js'
import { type UriQueryMap } from './WebTypes'

type PluginViewParams = {|
  // The GUI plugin we are showing the user:
  plugin: GuiPlugin,

  // Set these to add stuff to the plugin URI:
  deepPath?: string,
  deepQuery?: UriQueryMap
|}
/**
 * Defines the acceptable route parameters for each scene key.
 */
export type ParamList = {
  // Top-level router:
  root: {||},
  login: {||},
  edge: {||},

  // Logged-in scenes:
  changeMiningFee: {|
    guiMakeSpendInfo: GuiMakeSpendInfo,
    maxSpendSet: boolean,
    onSubmit: (networkFeeOption: FeeOption, customNetworkFee: JsonObject) => void,
    wallet: EdgeCurrencyWallet
  |},
  changePassword: {||},
  changePin: {||},
  controlPanel: {||},
  createWalletAccountSelect: {|
    accountName: string,
    existingWalletId?: string,
    selectedFiat: GuiFiatType,
    selectedWalletType: CreateWalletType
  |},
  createWalletAccountSetup: {|
    accountHandle?: string,
    existingWalletId?: string,
    isReactivation?: boolean,
    selectedFiat: GuiFiatType,
    selectedWalletType: CreateWalletType
  |},
  createWalletChoice: {|
    selectedWalletType: CreateWalletType
  |},
  createWalletImport: {|
    selectedWalletType: CreateWalletType
  |},
  createWalletName: {|
    cleanedPrivateKey?: string,
    selectedFiat: GuiFiatType,
    selectedWalletType: CreateWalletType
  |},
  createWalletReview: {|
    cleanedPrivateKey?: string, // for creating wallet from import private key
    selectedFiat: GuiFiatType,
    selectedWalletType: CreateWalletType,
    walletName: string
  |},
  createWalletSelectCrypto: {||},
  createWalletSelectFiat: {|
    selectedWalletType: CreateWalletType,
    cleanedPrivateKey?: string
  |},
  currencyNotificationSettings: {|
    currencyInfo: EdgeCurrencyInfo
  |},
  currencySettings: {|
    currencyInfo: EdgeCurrencyInfo
  |},
  defaultFiatSetting: {||},
  edgeLogin: {||},
  editToken: {|
    currencyCode?: string,
    displayName?: string,
    multiplier?: string,
    networkLocation?: JsonObject,
    tokenId?: string, // Acts like "add token" if this is missing
    walletId: string
  |},
  exchange: {||},
  exchangeQuote: {|
    swapInfo: GuiSwapInfo,
    onApprove: () => void
  |},
  exchangeQuoteProcessing: {||},
  exchangeScene: {||},
  exchangeSettings: {||},
  exchangeSuccess: {||},
  fioAddressDetails: {|
    fioAddressName: string,
    bundledTxs: number
  |},
  fioAddressList: {||},
  fioAddressRegister: {||},
  fioAddressRegisterSelectWallet: {|
    fioAddress: string,
    selectedWallet: EdgeCurrencyWallet,
    selectedDomain: FioDomain,
    isFallback?: boolean
  |},
  fioAddressRegisterSuccess: {|
    fioName: string,
    expiration?: string
  |},
  fioAddressSettings: {|
    fioWallet: EdgeCurrencyWallet,
    fioAddressName: string,
    bundledTxs?: number,
    showAddBundledTxs?: boolean,
    refreshAfterAddBundledTxs?: boolean
  |},
  fioConnectToWalletsConfirm: {|
    fioWallet: EdgeCurrencyWallet,
    fioAddressName: string,
    walletsToConnect: FioConnectionWalletItem[],
    walletsToDisconnect: FioConnectionWalletItem[]
  |},
  fioDomainConfirm: {|
    fioName: string,
    paymentWallet: EdgeCurrencyWallet,
    fee: number,
    ownerPublicKey: string
  |},
  fioDomainRegister: {||},
  fioDomainRegisterSelectWallet: {|
    fioDomain: string,
    selectedWallet: EdgeCurrencyWallet
  |},
  fioDomainSettings: {|
    fioWallet: EdgeCurrencyWallet,
    fioDomainName: string,
    isPublic: boolean,
    expiration: string,
    showRenew?: boolean
  |},
  fioNameConfirm: {|
    fioName: string,
    paymentWallet: EdgeCurrencyWallet,
    fee: number,
    ownerPublicKey: string
  |},
  fioRequestConfirmation: {|
    amounts: ExchangedFlipInputAmounts
  |},
  fioRequestList: {||},
  fioRequestApproved: {|
    edgeTransaction: EdgeTransaction,
    thumbnailPath?: string
  |},
  fioSentRequestDetails: {|
    selectedFioSentRequest: FioRequest
  |},
  fioStakingChange: {
    change: 'add' | 'remove',
    currencyCode: string,
    walletId: string
  },
  fioStakingOverview: {
    currencyCode: string,
    walletId: string
  },
  guiPluginEnterAmount: {|
    headerTitle: string,
    onSubmit: (response: FiatPluginEnterAmountResponse) => Promise<void>,
    label1: string,
    label2: string,
    onChangeText: (fieldNum: number, value: string) => Promise<void>,
    convertValue: (sourceFieldNum: number, value: string) => Promise<string | void>,
    getMethods?: (methods: FiatPluginGetMethodsResponse) => void,
    initialAmount1?: string,
    headerIconUri?: string
  |},
  loanDashboard: {||},
  loanDetails: {|
    loanAccountId: string
  |},
  loanCreate: {|
    borrowEngine: BorrowEngine,
    borrowPlugin: BorrowPlugin
  |},
  loanCreateConfirmation: {|
    borrowEngine: BorrowEngine,
    borrowPlugin: BorrowPlugin,
    destTokenId: string,
    destWallet: EdgeCurrencyWallet,
    isDestBank: boolean,
    nativeDestAmount: string,
    nativeSrcAmount: string,
    srcTokenId?: string,
    srcWallet: EdgeCurrencyWallet
  |},
  loanClose: {|
    loanAccountId: string
  |},
  loanAddCollateralScene: {|
    loanAccountId: string
  |},
  loanBorrowMoreScene: {|
    loanAccountId: string
  |},
  loanWithdrawCollateralScene: {|
    loanAccountId: string
  |},
  loanRepayScene: {|
    loanAccountId: string
  |},
  loanStatus: {|
    actionQueueId: string
  |},
  loanCreateStatus: {|
    actionQueueId: string
  |},
  loanDetailsStatus: {|
    actionQueueId: string
  |},
  manageTokens: {|
    walletId: string
  |},
  notificationSettings: {||},
  otpRepair: {|
    otpError: OtpError
  |},
  otpSetup: {||},
  passwordRecovery: {||},
  pluginListBuy: {| direction: 'buy' |},
  pluginListSell: {| direction: 'sell' |},
  pluginViewBuy: PluginViewParams,
  pluginViewSell: PluginViewParams,
  pluginView: PluginViewParams,
  promotionSettings: {||},
  request: {||},
  scan: {|
    data?: 'sweepPrivateKey' | 'loginQR'
  |}, // TODO
  securityAlerts: {||},
  send: {|
    allowedCurrencyCodes?: string[],
    guiMakeSpendInfo?: GuiMakeSpendInfo,
    selectedWalletId?: string,
    selectedCurrencyCode?: string,
    isCameraOpen?: boolean,
    lockTilesMap?: {
      address?: boolean,
      wallet?: boolean,
      amount?: boolean
    },
    hiddenTilesMap?: {
      address?: boolean,
      amount?: boolean,
      fioAddressSelect?: boolean
    },
    infoTiles?: Array<{ label: string, value: string }>
  |},
  settingsOverview: {||},
  settingsOverviewTab: {||},
  spendingLimits: {||},
  stakeModify: {|
    walletId: string,
    stakePolicy: StakePolicy,
    stakePosition: StakePosition,
    modification: $PropertyType<ChangeQuoteRequest, 'action'>
  |},
  stakeClaim: { walletId: string, stakePolicy: StakePolicy },
  stakeOptions: { walletId: string, currencyCode: string },
  stakeOverview: { walletId: string, stakePolicy: StakePolicy },
  termsOfService: {||},
  testScene: {||},
  transactionDetails: {|
    edgeTransaction: EdgeTransaction,
    thumbnailPath?: string
  |},
  transactionList: {||},
  transactionsExport: {|
    sourceWallet: EdgeCurrencyWallet,
    currencyCode: string
  |},
  walletList: {||},
  walletListScene: {||},
  wcConnections: {||},
  wcDisconnect: {| wcConnectionInfo: WcConnectionInfo |},
  wcConnect: {|
    uri: string
  |}
}

/**
 * The global `Actions` object for navigation.
 */
export const Actions = {
  get currentParams(): any {
    // $FlowFixMe
    return Flux.Actions.currentParams
  },
  get currentScene(): $Keys<ParamList> {
    // $FlowFixMe
    return Flux.Actions.currentScene
  },

  drawerClose() {
    // $FlowFixMe
    Flux.Actions.drawerClose()
  },
  drawerOpen() {
    // $FlowFixMe
    Flux.Actions.drawerOpen()
  },

  jump<Name extends $Keys<ParamList>>(name: Name, params: $ElementType<ParamList, Name>): void {
    // $FlowFixMe
    Flux.Actions.jump(name, { route: { name, params } })
  },
  push<Name extends $Keys<ParamList>>(name: Name, params: $ElementType<ParamList, Name>): void {
    // $FlowFixMe
    Flux.Actions.push(name, { route: { name, params } })
  },
  replace<Name extends $Keys<ParamList>>(name: Name, params: $ElementType<ParamList, Name>): void {
    // $FlowFixMe
    Flux.Actions.replace(name, { route: { name, params } })
  },

  refresh(params: any): void {
    // $FlowFixMe
    Flux.Actions.refresh({ route: { name: Flux.Actions.currentScene, params } })
  },

  pop(): void {
    // $FlowFixMe
    Flux.Actions.pop()
  },
  popTo(name: $Keys<ParamList>): void {
    // $FlowFixMe
    Flux.Actions.popTo(name)
  }
}

type NavigationEvent = 'didBlur' | 'didFocus' | 'willBlur' | 'willFocus'

/**
 * The type of the `navigation` prop passed to each scene.
 */
export type NavigationProp<Name extends $Keys<ParamList>> = {
  // Whether this scene is in the foreground:
  addListener: (event: NavigationEvent, callback: () => void) => () => void,
  isFocused: () => boolean,

  // Going places:
  navigate: <Name extends $Keys<ParamList>>(name: Name, params: $ElementType<ParamList, Name>) => void,
  push: <Name extends $Keys<ParamList>>(name: Name, params: $ElementType<ParamList, Name>) => void,
  replace: <Name extends $Keys<ParamList>>(name: Name, params: $ElementType<ParamList, Name>) => void,
  setParams: (params: $ElementType<ParamList, Name>) => void,

  // Returning:
  goBack: () => void,
  pop: () => void,
  popToTop: () => void,

  // Drawer:
  closeDrawer: () => void,
  openDrawer: () => void,
  toggleDrawer: () => void,

  // Internals nobody should need to touch:
  state: mixed
}

/**
 * The type of the `route` prop passed to each scene.
 */
export type RouteProp<Name extends $Keys<ParamList>> = {
  name: Name,
  params: $ElementType<ParamList, Name>
}

/**
 * Adjusts the navigation prop to match the type definitions above.
 */
export function withNavigation<Props>(Component: React.ComponentType<Props>): React.StatelessFunctionalComponent<Props> {
  function WithNavigation(props: any) {
    const navigation: NavigationProp<'edge'> = {
      addListener(event, callback) {
        const remover = props.navigation.addListener(event, callback)
        return () => remover.remove()
      },
      isFocused() {
        return props.navigation.isFocused()
      },

      navigate(name, params) {
        props.navigation.navigate(name, { route: { name, params } })
      },
      push(name, params) {
        props.navigation.push(name, { route: { name, params } })
      },
      replace(...a) {
        // prev, next, params) {
        props.navigation.replace(...a) // prev, next, next, { route: { params } })
      },
      setParams(params) {
        props.navigation.setParams({ route: { name: Actions.currentScene, params } })
      },

      goBack() {
        props.navigation.goBack()
      },
      pop() {
        props.navigation.pop()
      },
      popToTop() {
        props.navigation.popToTop()
      },

      closeDrawer() {
        props.navigation.closeDrawer()
      },
      openDrawer() {
        props.navigation.openDrawer()
      },
      toggleDrawer() {
        props.navigation.toggleDrawer()
      },

      get state() {
        return props.navigation.state
      }
    }

    return <Component {...props} navigation={navigation} />
  }
  const displayName = Component.displayName ?? Component.name ?? 'Component'
  WithNavigation.displayName = `WithNavigation(${displayName})`
  return WithNavigation
}
