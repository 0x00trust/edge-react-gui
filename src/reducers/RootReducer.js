// @flow

import { type Reducer, combineReducers } from 'redux'

import { type ExchangeRatesState, exchangeRates } from '../modules/ExchangeRates/reducer.js'
import { type UiState, ui } from '../modules/UI/reducer.js'
import { type Action } from '../types/reduxTypes.js'
import { type ContactsState, contacts } from './ContactsReducer.js'
import { type CoreState, core } from './CoreReducer.js'
import { type CryptoExchangeState, cryptoExchange } from './CryptoExchangeReducer.js'
import { type PermissionsState, permissions } from './PermissionsReducer.js'

export type RootState = {
  // Next username to auto-fill at the login screen, or blank if none:
  +nextUsername: string | null,

  // Nested reducers:
  +contacts: ContactsState,
  +core: CoreState,
  +cryptoExchange: CryptoExchangeState,
  +exchangeRates: ExchangeRatesState,
  +permissions: PermissionsState,
  +ui: UiState
}

export const rootReducer: Reducer<RootState, Action> = combineReducers({
  nextUsername (state: string | null = null, action: Action): string | null {
    switch (action.type) {
      case 'DEEP_LINK_RECEIVED':
        return null
      case 'LOGOUT': {
        const { username = null } = action.data
        return username
      }
    }
    return state
  },

  // Nested reducers:
  contacts,
  core,
  cryptoExchange,
  exchangeRates,
  permissions,
  ui
})
