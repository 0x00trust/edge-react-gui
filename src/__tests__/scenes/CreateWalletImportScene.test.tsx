import { describe, expect, it, jest } from '@jest/globals'
import * as React from 'react'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import { createStore } from 'redux'

import { CreateWalletImportScene } from '../../components/scenes/CreateWalletImportScene'
import { rootReducer } from '../../reducers/RootReducer'
import { RouteProp } from '../../types/routerTypes'
import { fakeNavigation } from '../../util/fake/fakeNavigation'

jest.mock('react-native-keyboard-aware-scroll-view', () => {
  const KeyboardAwareScrollView = (blob: { children: React.ReactNode }) => blob.children
  return { KeyboardAwareScrollView }
})

// Jest doesn't like direct SVG imports
jest.mock('../../assets/images/import-key-icon.svg', () => 'ImportKeySvg')

describe('CreateWalletImportScene', () => {
  const mockState: any = {
    core: {
      account: {
        currencyConfig: {
          bitcoin: {
            importKey: () => {}
          }
        }
      }
    }
  }
  const store = createStore(rootReducer, mockState)

  it('should render with loading props', () => {
    const navigation = fakeNavigation
    const route: RouteProp<'createWalletImport'> = {
      name: 'createWalletImport',
      params: {
        createWalletList: [
          {
            key: `create-wallet:bitcoin-bip49-bitcoin`,
            currencyCode: 'BTC',
            displayName: 'Bitcoin',
            pluginId: 'bitcoin',
            walletType: 'wallet:bitcoin-bip49'
          }
        ],
        walletNames: { 'create-wallet:bitcoin-bip49-bitcoin': 'My Bitcoin' },
        fiatCode: 'USD'
      }
    }

    const actual = renderer.create(
      <Provider store={store}>
        <CreateWalletImportScene navigation={navigation} route={route} />
      </Provider>
    )

    expect(actual).toMatchSnapshot()
  })
})
