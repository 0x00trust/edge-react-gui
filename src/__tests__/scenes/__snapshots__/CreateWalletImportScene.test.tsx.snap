// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateWalletImportScene should render with loading props 1`] = `
<BVLinearGradient
  colors={
    Array [
      4279050565,
      4278994026,
    ]
  }
  endPoint={
    Object {
      "x": 1,
      "y": 0,
    }
  }
  locations={null}
  startPoint={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "alignItems": "stretch",
        "bottom": 0,
        "flexDirection": "column",
        "justifyContent": "flex-start",
        "left": 0,
        "maxHeight": 1270,
        "padding": 0,
        "position": "absolute",
        "right": 0,
        "top": 64,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "justifyContent": "center",
            "marginLeft": 22,
            "paddingBottom": 22,
          },
          Object {
            "marginTop": 22,
          },
          undefined,
        ]
      }
    >
      <Text
        adjustsFontSizeToFit={true}
        minimumFontScale={0.65}
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Quicksand-Regular",
              "fontSize": 22,
              "includeFontPadding": false,
            },
            Object {
              "fontFamily": "Quicksand-Medium",
              "fontSize": 27,
            },
            null,
          ]
        }
      >
        Import Wallet
      </Text>
    </View>
    <View
      style={
        Object {
          "marginBottom": 11,
          "marginLeft": 22,
        }
      }
    />
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
          "marginVertical": 22,
        }
      }
    >
      <ImportKeySvg
        color="#66EDA8"
        height={0}
        width={0}
      />
    </View>
    <Text
      adjustsFontSizeToFit={true}
      minimumFontScale={0.65}
      numberOfLines={2}
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Quicksand-Regular",
            "fontSize": 22,
            "includeFontPadding": false,
          },
          Object {
            "color": "#FFFFFF",
            "fontSize": 22,
            "marginBottom": 22,
            "marginHorizontal": 11,
            "marginTop": 11,
            "paddingHorizontal": 22,
            "textAlign": "center",
          },
          null,
        ]
      }
    >
      Enter your private seed, private key, or active key to verify and restore the associated wallet
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "rgba(255, 255, 255, 0)",
            "borderRadius": 11,
            "justifyContent": "center",
            "minHeight": 67,
            "paddingHorizontal": 22,
          },
          Object {
            "flexGrow": 0,
            "marginBottom": 28,
            "marginLeft": 17,
            "marginRight": 17,
            "marginTop": 22,
            "paddingLeft": 22,
            "paddingRight": 65,
            "paddingVertical": 0,
          },
        ]
      }
    >
      <View
        animatedStyle={
          Object {
            "value": Object {
              "borderColor": "rgba(164, 199, 223, 1)",
              "right": 11,
            },
          }
        }
        collapsable={false}
        style={
          Object {
            "borderColor": "rgba(164, 199, 223, 1)",
            "borderTopWidth": 1,
            "bottom": 0,
            "left": 22,
            "position": "absolute",
            "right": 11,
          }
        }
      />
      <View
        animatedStyle={
          Object {
            "value": Object {
              "borderColor": "rgba(164, 199, 223, 1)",
            },
          }
        }
        collapsable={false}
        style={
          Object {
            "borderBottomLeftRadius": 11,
            "borderBottomWidth": 1,
            "borderColor": "rgba(164, 199, 223, 1)",
            "borderLeftWidth": 1,
            "borderRightWidth": 0,
            "borderTopLeftRadius": 11,
            "borderTopWidth": 1,
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "top": 0,
            "width": 22,
          }
        }
      />
      <View
        animatedStyle={
          Object {
            "value": Object {
              "borderColor": "rgba(164, 199, 223, 1)",
            },
          }
        }
        collapsable={false}
        style={
          Object {
            "borderBottomRightRadius": 11,
            "borderBottomWidth": 1,
            "borderColor": "rgba(164, 199, 223, 1)",
            "borderLeftWidth": 0,
            "borderRightWidth": 1,
            "borderTopRightRadius": 11,
            "borderTopWidth": 1,
            "bottom": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 22,
          }
        }
      />
      <View
        animatedStyle={
          Object {
            "value": Object {
              "borderColor": "rgba(164, 199, 223, 1)",
              "left": 22,
            },
          }
        }
        collapsable={false}
        style={
          Object {
            "borderColor": "rgba(164, 199, 223, 1)",
            "borderTopWidth": 1,
            "left": 22,
            "position": "absolute",
            "right": 22,
            "top": 0,
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "height": 67,
              "justifyContent": "center",
              "left": 0,
              "paddingHorizontal": 22,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "paddingLeft": 22,
              "paddingRight": 65,
            },
          ]
        }
      >
        <Text
          animatedStyle={
            Object {
              "value": Object {
                "color": "rgba(164, 199, 223, 1)",
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateY": -0,
                  },
                  Object {
                    "scale": 1,
                  },
                ],
              },
            }
          }
          collapsable={false}
          numberOfLines={1}
          onLayout={[Function]}
          style={
            Object {
              "alignSelf": "flex-start",
              "color": "rgba(164, 199, 223, 1)",
              "fontFamily": "Quicksand-Regular",
              "fontSize": 22,
              "padding": 0,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": -0,
                },
                Object {
                  "scale": 1,
                },
              ],
            }
          }
        >
          Private Key or Private Seed
        </Text>
      </View>
      <Text
        animatedStyle={
          Object {
            "value": Object {
              "opacity": 0,
            },
          }
        }
        collapsable={false}
        numberOfLines={1}
        style={
          Object {
            "bottom": -28,
            "color": "#E85466",
            "fontFamily": "Quicksand-Regular",
            "fontSize": 17,
            "left": 28,
            "opacity": 0,
            "position": "absolute",
          }
        }
      />
      <Text
        animatedStyle={
          Object {
            "value": Object {
              "color": "rgba(164, 199, 223, 1)",
              "opacity": 0,
              "transform": Array [
                Object {
                  "translateX": 6,
                },
                Object {
                  "scale": 0,
                },
              ],
            },
          }
        }
        collapsable={false}
        numberOfLines={1}
        onLayout={[Function]}
        style={
          Object {
            "bottom": -10,
            "color": "rgba(164, 199, 223, 1)",
            "fontFamily": "Quicksand-Regular",
            "fontSize": 17,
            "opacity": 0,
            "position": "absolute",
            "right": 28,
            "transform": Array [
              Object {
                "translateX": 6,
              },
              Object {
                "scale": 0,
              },
            ],
          }
        }
      >
        
      </Text>
      <TextInput
        autoCapitalize="none"
        autoCorrect={false}
        autoFocus={true}
        editable={true}
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        returnKeyType="next"
        secureTextEntry={false}
        selectionColor="#FFFFFF"
        style={
          Array [
            Object {
              "alignSelf": "stretch",
              "color": "#FFFFFF",
              "fontFamily": "Quicksand-Regular",
              "fontSize": 22,
              "padding": 0,
            },
            Object {
              "flexGrow": 0,
            },
          ]
        }
        textAlignVertical="top"
        value=""
      />
    </View>
    <View
      onLayout={[Function]}
    >
      <View
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "elevation": 0,
            "opacity": 1,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
      >
        <BVLinearGradient
          colors={
            Array [
              16777215,
              16777215,
            ]
          }
          endPoint={
            Object {
              "x": 1,
              "y": 1,
            }
          }
          locations={null}
          startPoint={
            Object {
              "x": 0,
              "y": 0,
            }
          }
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderColor": "#66EDA8",
                "borderRadius": 6,
                "borderWidth": 1,
                "flexDirection": "row",
                "justifyContent": "center",
                "minHeight": 67,
                "minWidth": 202,
              },
              Object {
                "alignSelf": "center",
                "marginBottom": 11,
                "marginLeft": 11,
                "marginRight": 11,
                "marginTop": 11,
                "opacity": 1,
                "paddingBottom": 0,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 0,
              },
              Object {
                "borderRadius": 6,
                "paddingLeft": 0,
                "paddingRight": 0,
              },
            ]
          }
        >
          <Text
            adjustsFontSizeToFit={true}
            minimumFontScale={0.75}
            numberOfLines={1}
            style={
              Object {
                "color": "#66EDA8",
                "fontFamily": "Quicksand-Regular",
                "fontSize": 22,
                "marginHorizontal": 0,
                "paddingBottom": 11,
                "paddingLeft": 17,
                "paddingRight": 17,
                "paddingTop": 11,
              }
            }
          >
            Next
          </Text>
        </BVLinearGradient>
      </View>
    </View>
  </View>
</BVLinearGradient>
`;
